doubleml/double_ml_ssm.py

```diff
@@ -0,0 +1,476 @@
+                                    est_params=self._get_params('ml_m'), method=self._predict_method['ml_m'],
+                                    return_models=return_models)
+            m_hat['targets'] = m_hat['targets'].astype(float)
+            _check_finite_predictions(m_hat['preds'], self._learner['ml_m'], 'ml_m', smpls)
+
+            # conditional outcome
+            g_hat_d1 = _dml_cv_predict(self._learner['ml_g'], x, y, smpls=smpls_d1_s1, n_jobs=n_jobs_cv,
+                                       est_params=self._get_params('ml_g_d1'), method=self._predict_method['ml_g'],
+                                       return_models=return_models)
+            g_hat_d1['targets'] = g_hat_d1['targets'].astype(float)
+            _check_finite_predictions(g_hat_d1['preds'], self._learner['ml_g'], 'ml_g_d1', smpls)
+
+            g_hat_d0 = _dml_cv_predict(self._learner['ml_g'], x, y, smpls=smpls_d0_s1, n_jobs=n_jobs_cv,
+                                       est_params=self._get_params('ml_g_d0'), method=self._predict_method['ml_g'],
+                                       return_models=return_models)
+            g_hat_d0['targets'] = g_hat_d0['targets'].astype(float)
+            _check_finite_predictions(g_hat_d0['preds'], self._learner['ml_g'], 'ml_g_d0', smpls)
+
+        else:
+            assert self._score == 'nonignorable'
+            # initialize nuisance predictions, targets and models
+            g_hat_d1 = {'models': None,
+                        'targets': np.full(shape=self._dml_data.n_obs, fill_value=np.nan),
+                        'preds': np.full(shape=self._dml_data.n_obs, fill_value=np.nan)
+                        }
+            g_hat_d0 = copy.deepcopy(g_hat_d1)
+            pi_hat = copy.deepcopy(g_hat_d1)
+            m_hat = copy.deepcopy(g_hat_d1)
+
+            # create strata for splitting
+            strata = self._dml_data.d.reshape(-1, 1) + 2 * self._dml_data.t.reshape(-1, 1)
+
+            # pi_hat - used for preliminary estimation of propensity score pi, overwritten in each iteration
+            pi_hat_prelim = {'models': None,
+                             'targets': [],
+                             'preds': []
+                             }
+
+            # calculate nuisance functions over different folds - nested cross-fitting
+            for i_fold in range(self.n_folds):
+                train_inds = smpls[i_fold][0]
+                test_inds = smpls[i_fold][1]
+
+                # start nested crossfitting - split training data into two equal parts
+                train_inds_1, train_inds_2 = train_test_split(
+                    train_inds, test_size=0.5, random_state=42, stratify=strata[train_inds]
+                )
+
+                s_train_1 = s[train_inds_1]
+                dx_train_1 = dx[train_inds_1, :]
+
+                # preliminary propensity score for selection
+                ml_pi_prelim = clone(self._learner['ml_pi'])
```

Comment:
These models are not saved right? I guess the ``pi_hat['models']`` are ``None``. (see e.g. https://github.com/mychaelka/doubleml-for-py/blob/fb6a38fd7bbe5632fda7473b48e333e224b5a22e/doubleml/double_ml_pq.py#L299)
If i am reading your [paper](https://arxiv.org/abs/2012.00745) correctly, these models are not refitted right? Then we might already remove the preliminary part of the name (this was just used in Quantile models for a 2-step procedure since the propensity score has to be refitted).

Reply:
You're right, I set the models to be saved (hopefully it's correct this way) and removed the preliminary part of the name (I used it to emphasize that nuisance function pi is estimated first on a different subsample). I also set the above and below mentioned defaults to False.

Reply:
Thank you. Looks fine to me. I will check it in detail later.